src/azure.ad/basicauth_inbound_handler.bal

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+import ballerina/http;
+import ballerina/lang.'string as sutils;
+
+# Configuration for azure active directory authenticator provider
+# 
+# + tenantId - Tenant ID for the active directory
+# + clientId - Client ID for the client credentials grant authentication
+# + clientSecret - Client secret for the client credentials grant authentication
+# + scopes - Scope(s) of the access request
+public type InboundAzureAdUserAuthenticatorProviderConfig record {|
```

Comment:
Shall we rename the type to `InboundUserAuthenticatorProviderConfig`. Since it `azure.ad` is in the module prefix, we don't need. WDYT?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerinax-azure.ad/pull/1/commits/a534cb1f06c82d058707850878f95cd1e405c41b